core/src/wallet/ripple/explorers/NodeRippleLikeBlockchainExplorer.cpp

```diff
@@ -0,0 +1,282 @@
+            return Future<Unit>::successful(unit);
+        }
+
+        Future<Bytes> NodeRippleLikeBlockchainExplorer::getRawTransaction(const String &transactionHash) {
+            NodeRippleLikeBodyRequest bodyRequest;
+            bodyRequest.setMethod("tx");
+            bodyRequest.pushParameter("trasnaction", transactionHash);
+            bodyRequest.pushParameter("binary", "true");
+            auto requestBody = bodyRequest.getString();
+            return _http->POST("", std::vector<uint8_t>(requestBody.begin(), requestBody.end()))
+                    .json().template map<Bytes>(getExplorerContext(), [](const HttpRequest::JsonResult &result) -> Bytes {
+                        auto &json = *std::get<1>(result);
+                        if (!json.IsObject() || !json.HasMember("result") ||
+                            !json["result"].IsObject()) {
+                            throw make_exception(api::ErrorCode::HTTP_ERROR, "Failed to get raw transaction, no (or malformed) field \"result\" in response");
+                        }
+                        //Is there a tx field ?
+                        auto resultObj = json["result"].GetObject();
+                        if (!resultObj.HasMember("tx") || !resultObj["tx"].IsString()) {
+                            throw make_exception(api::ErrorCode::HTTP_ERROR, "Failed to get raw transaction, no (or malformed) field \"tx\" in response");
+                        }
+                        return Bytes(hex::toByteArray(std::string(resultObj["tx"].GetString(), resultObj["tx"].GetStringLength())));
+                    });
+        }
+
+        Future<String> NodeRippleLikeBlockchainExplorer::pushTransaction(const std::vector<uint8_t> &transaction) {
+            return pushLedgerApiTransaction(transaction);
+        }
+
+        FuturePtr<RippleLikeBlockchainExplorer::TransactionsBulk>
+        NodeRippleLikeBlockchainExplorer::getTransactions(const std::vector<std::string> &addresses,
+                                                          Option<std::string> fromBlockHash,
+                                                          Option<void *> session) {
+            if (addresses.size() != 1) {
+                throw make_exception(api::ErrorCode::INVALID_ARGUMENT,
+                                     "Can only get transactions for 1 address from Ripple Node, but got {} addresses", addresses.size());
+            }
+            NodeRippleLikeBodyRequest bodyRequest;
+            bodyRequest.setMethod("account_tx");
+            bodyRequest.pushParameter("account", addresses[0]);
+            auto requestBody = bodyRequest.getString();
+            return _http->POST("", std::vector<uint8_t>(requestBody.begin(), requestBody.end()))
+                    .template json<TransactionsBulk, Exception>(
+                            LedgerApiParser<TransactionsBulk, RippleLikeTransactionsBulkParser>())
+                    .
```

Comment:
![](https://media.tenor.com/images/4ae0560567ebab7440ae6cf7f8a82848/tenor.gif)

Reply:
WTF happened here? 😆

Reply:
Oh god damn it ! I think I used Clion's code formatter and he f***ed up everything 😆